README.md

```diff
@@ -46,7 +46,69 @@ ActiveRecord models that use this connection will now be connecting to the confi
 
 ## Testing
 
-To run the tests, you'll need the ODBC driver as well as the connection adapter for each database against which you're trying to test. Then run `DSN=MyDatabaseDSN bundle exec rake test` and the test suite will be run by connecting to your database.
+To run the tests, you'll need the ODBC driver as well as the connection adapter for each database against which you're trying to test. You'll then need to set either the `CONN_STR` or `DSN` env var to point at the database and driver you're using.
+
+1. Install ruby 3.2
+
+   - If you don't already have ruby 3.2 installed locally, one option is using rbenv via the following:
+
+      1. Install [rbenv](https://github.com/rbenv/rbenv?tab=readme-ov-file#homebrew) via homebrew
+      2. `rbenv install 3.2.0`
+      3. `rbenv local 3.2.0`
+
+   - Note: if you are using nix for local development of other apps, the version of libyaml installed by this method can interfere with the building of native extensions for those apps. To fix that, you'll need to run `brew uninstall --ignore-dependencies libyaml` and re-run your bundle install inside of that nix shell. You'll then need to reinstall libyaml via homebrew if you want to work on this gem.
+
+2. Run `bundle install` in this directory to install the relevant gems
+
+3. Install the odbc driver for your database
+
+    - Postgres example:
+
+        1. Install postgres via [installer or homebrew](https://www.postgresql.org/download/macosx/) (if not already installed)
+
+        2. Install the driver via [homebrew](https://formulae.brew.sh/formula/psqlodbc): `brew install psqlodbc`
+
+        3. Create a `~/.odbcinst.ini` pointing to your driver:
+
+            ```
+            [PostgreSQL ANSI]
+            Description     = Postgres connection
+            Driver          = /opt/homebrew/Cellar/psqlodbc/17.00.0003/lib/psqlodbcw.so
+            UsageCount      = 1
+            Debug           = 0
+            CommLog         = 1
+            ```
+
+4. Configure your database connection
+
+   - Using `CONN_STR`:
+
+     1. Create a postgres database to use or get the configuration details for an existing one
+
+     2. Set a CONN_STR with the connection details: `set CONN_STR=DRIVER={PostgreSQL ANSI};SERVER=127.0.0.2;PORT=5432;DATABASE=foo_development;UID=USERNAME_HERE;password=PASSWORD_HERE`
+
+   - Using `DSN`
```

Comment:
I don't have a ton of confidence in this approach because I used the `CONN_STR`

Tests do pass locally if I use the `DSN` env var but I don't understand how because we're not providing credentials...